tensornetwork/contractors/custom_path_solvers/solve_functs.py

```diff
@@ -0,0 +1,470 @@
+  """  
+  N = log_adj.shape[0]
+  log_adj = log_adj.reshape(N,N)
+  orders = np.zeros([2,0],dtype=int)
+  costs = 0;
+  
+  for t in range(N-1):
+    # compute tensor dims and costs
+    N = log_adj.shape[0]
+    dims = np.sum(log_adj,axis=0).reshape(N)
+    comb_dims = np.add.outer(dims,dims) 
+    single_cost = comb_dims - log_adj
+    
+    # penalize trivial contractions and self-contractions
+    triv_conts = (log_adj < 0.001) # small fudge factor 
+    trimmed_costs = single_cost + max(single_cost.flatten())*triv_conts
+    trimmed_costs = trimmed_costs + max(trimmed_costs.flatten())*np.eye(N)
+    
+    # find best contraction
+    tensors_to_contract = np.divmod(np.argmin(trimmed_costs),N)
+    i = max(tensors_to_contract)
+    j = min(tensors_to_contract)
+    
+    # build new log adjacency
+    log_adj[j,j] = log_adj[j,j] - 2*log_adj[j,i] 
+    log_adj[j,:] = log_adj[j,:] + log_adj[i,:]
+    log_adj[:,j] = log_adj[:,j] + log_adj[:,i]
+    log_adj = np.delete(log_adj, i, axis=0)
+    log_adj = np.delete(log_adj, i, axis=1)
+    
+    # build new orders
+    orders = np.hstack((orders,np.asarray(tensors_to_contract).reshape(2,1)))
+    
+    # tally the cost
+    costs = np.log10(10**costs + 10**single_cost[i,j])
+  
+  return orders, costs
+
+#------------------------------------------------------------------------#
+def full_solve_complete(log_adj: np.ndarray,
+                        cost_bound: Optional[int] = None,
+                        max_kept: Optional[int] = None):
+  """
+  Solve for optimal contraction path of a network encoded as a log-adjacency 
+  matrix via a full search. 
+  Args:
+    log_adj: the log10 of the adjacency matrix of the network.
+    cost_bound: upper cost threshold for discarding paths, in log10(FLOPS).
+    max_kept: bound for the total number of paths to retain.
+  Returns:
+    np.ndarray: the cheapest contraction order found.
+    float: the cost of the network contraction, given as log10(total_FLOPS).
+    bool: specifies if contraction order is guaranteed optimal.
+  """  
+  # start by trying both greedy algorithms
+  order0, cost0 = greedy_size_solve(log_adj.copy())
+  order1, cost1 = greedy_cost_solve(log_adj.copy())
+  if (cost0 < cost1):
+    order_greedy = order0
+    cost_greedy = cost0
+  else:
+    order_greedy = order1
+    cost_greedy = cost1
+  
+  if (max_kept == 1):
+    # return results from greedy
+    order_was_found = False
+  else:
+    # initialize arrays
+    N = log_adj.shape[0]
+    costs = np.zeros([1,1])
+    groups =  np.array(2**np.arange(N),dtype=np.uint64).reshape(N,1)
+    orders = np.zeros([2,0,1],dtype=int)
+    
+    # try full algorithm (using cost_bound from greedy)
+    cost_bound = cost_greedy + 1e-4 #small fudge factor
+    total_truncated = 0
+    order_was_found = True
+    for t in range(N-1):
+      log_adj,costs,groups,orders,num_truncated = full_solve_single(log_adj,costs,groups,orders,cost_bound=cost_bound,max_kept=max_kept)
```

Comment:
length of lines shoule be less than 80 characters.

Easiest fix here is to add a newline after the `(`.

Reply:
thanks for the reminder, should be now fixed in all files